apps/explorer/src/components/TransactionBlocksForAddress/TransactionBlocksForAddress.tsx

```diff
@@ -148,7 +148,10 @@ function TransactionBlocksForAddress({
 								filterValue,
 							});
 						}}
-						hasNext={Boolean(hasNextPage) && Boolean(data?.pages[currentPage])}
+						hasNext={
+							(Boolean(hasNextPage) && Boolean(data?.pages[currentPage])) ||
+							currentPage < (data?.pages.length ?? 0) - 1
```

Comment:
What's the case this is fixing? Could you explain a bit the logic here?

Reply:
`hasNextPage` will stay `false` after hitting the last page, even if you go back to previous page. This helps to make sure that going back will allow user to go forth again.